pkg/controllers/provisioning/provisioner.go

```diff
@@ -131,6 +131,14 @@ func (p *Provisioner) Reconcile(ctx context.Context, _ reconcile.Request) (resul
 	if len(results.NewNodeClaims) == 0 {
 		return reconcile.Result{}, nil
 	}
+	np , err := nodepoolutil.Get(ctx, p.kubeClient, results.NewNodeClaims[0].OwnerKey)
```

Comment:
We shouldn't grab the NodePool here. I think the right place to do this is in the filtering logic on L212, where we already remove NodePools that are deleting, we can just add the logic that filters out NodePools that aren't ready